### PR TITLE
Add onSystemContext to cancel interaction calls

### PR DESCRIPTION
**Pull request by JackLivio**
Adds onSystemContext(MAIN) calls to cancel interaction requests.

### CODE REVIEW

**Comment by iCollin** [comment]
my concern is what if the HMI was not in MAIN context before the interaction? For ex Perform Interaction or a SubtleAlert in LIMITED while another app is active.

**Comment by JackLivio** [comment]
SystemContext of MAIN still applies to app's that are in HMI_LEVEL LIMITED/BACKGROUND

File: src/js/Controllers/UIController.js [hunk 1]
```diff
@@ -21,6 +21,20 @@ import store from '../store'
 import sdlController from './SDLController'
 import SubmenuDeepFind from '../Utils/SubMenuDeepFind'
 
+const getNextSystemContext = () => {
+    const state = store.getState();
+    const activeApp = state.activeApp;
+    const pathName = window.location.pathname;
```
Comment (iCollin) [inline src/js/Controllers/UIController.js]:
```suggestion
    const pathName = window.location.hash;
```

you could potentially use === with the hash property

File: src/js/Controllers/UIController.js [hunk 2]
```diff
@@ -21,6 +21,20 @@ import store from '../store'
 import sdlController from './SDLController'
 import SubmenuDeepFind from '../Utils/SubMenuDeepFind'
 
+const getNextSystemContext = () => {
+    const state = store.getState();
+    const activeApp = state.activeApp;
+    const pathName = window.location.href;
```
Comment (iCollin) [inline src/js/Controllers/UIController.js:27]:
```suggestion
    const pathName = window.location.hash;
```

could use `===` with hash property

**Review by iCollin — APPROVED** [review]

**Pull request merged**